packages/backend/server/src/plugins/indexer/__tests__/providers/elasticsearch.spec.ts

```diff
@@ -0,0 +1,1407 @@
+        additional: 'additional8',
+        markdown_preview: 'markdownPreview8',
+        created_by_user_id: 'userId8',
+        updated_by_user_id: 'userId8',
+        created_at: new Date('2025-03-08T06:04:13.278Z'),
+        updated_at: new Date('2025-03-08T06:04:13.278Z'),
+      },
+      {
+        workspace_id: 'workspaceId1',
+        doc_id: 'docId2-affine',
+        block_id: 'blockId8',
+        content: 'AFFiNE 是一个基于云端的笔记应用',
+        flavour: 'flavour8',
+        ref_doc_id: 'docId1',
+        ref: [
+          '{"docId":"docId1","mode":"page"}',
+          '{"docId":"docId2","mode":"page"}',
+        ],
+        parent_flavour: 'parentFlavour8',
+        parent_block_id: 'parentBlockId8',
+        additional: 'additional8',
+        markdown_preview: 'markdownPreview8',
+        created_by_user_id: 'userId8',
+        updated_by_user_id: 'userId8',
+        created_at: new Date('2025-03-08T06:04:13.278Z'),
+        updated_at: new Date('2025-03-08T06:04:13.278Z'),
+      },
+    ],
+    {
+      refresh: true,
+    }
+  );
+  const blocks = await readFile(
+    path.join(import.meta.dirname, '../test-blocks.json'),
+    'utf-8'
+  );
+  // @ts-expect-error access protected method
+  await searchProvider.requestBulk(
+    SearchTable.block,
+    blocks.trim().split('\n'),
+    {
+      // make sure the data is visible to search
+      refresh: 'true',
+    }
+  );
+  const docs = await readFile(
+    path.join(import.meta.dirname, '../test-docs.json'),
+    'utf-8'
+  );
+  // @ts-expect-error access protected method
+  await searchProvider.requestBulk(SearchTable.doc, docs.trim().split('\n'), {
+    refresh: 'true',
+  });
+});
+
+test.after.always(async () => {
+  await searchProvider.deleteByQuery(
+    SearchTable.block,
+    {
+      term: {
+        workspace_id: workspace.id,
+      },
+    },
+    {
+      refresh: true,
+    }
+  );
+  await searchProvider.deleteByQuery(
+    SearchTable.doc,
+    {
+      term: {
+        workspace_id: workspace.id,
+      },
+    },
+    {
+      refresh: true,
+    }
+  );
+  await module.close();
+});
+
+test('should provider is elasticsearch', t => {
+  t.is(searchProvider.provider, SearchProviderName.Elasticsearch);
+});
+
+// #region write
+
+test('should write document work', async t => {
+  const docId = randomUUID();
+  await searchProvider.write(
+    SearchTable.block,
+    [
+      {
+        workspace_id: workspace.id,
+        doc_id: docId,
+        content: 'hello world',
+        flavour: 'affine:page',
+        created_by_user_id: user.id,
+        updated_by_user_id: user.id,
+        created_at: new Date(),
+        updated_at: new Date(),
+      },
+    ],
+    {
+      refresh: true,
+    }
+  );
+  let result = await searchProvider.search(SearchTable.block, {
+    _source: ['workspace_id', 'doc_id'],
+    query: { match: { doc_id: docId } },
+    fields: ['flavour', 'block_id', 'content', 'ref_doc_id'],
+    sort: ['_score'],
+  });
+  t.is(result.nodes.length, 1);
```

Comment:
please add empty lines between preparation/actions/asserts, to make tests clearer